### ISSUE TITLE
[question] How do I pass/catch params in request?

### ISSUE THREAD

**Issue by MilosStanic** [SaifurRahmanMohsin/RESTful-Plugin#1]
Hi, 

A beautifully simple plugin. 
However, I'm interested in customizing responses. 

For example, if I have 'index' verb passing to my model 'Test', and I want to get only a particular Test, not all tests, and with it also Questions that are related to that test via Question model, how would I do that?

So, I want to pass a request to 'myserver/api/v1/tests/23'

And I want to get a Test with ID 23, and all belonging questions in a form

``` 
[
    { id: 23,
      title: 'My test 23 title',
      questions: [
                           { id: 253,
                             text: 'What is the boiling point of water?',
                             answer: '100 degrees Celsius'
                           },
                           { id: 255,
                             text: 'What is the freezing point of water?', 
                             answer: '0 degrees Celsius'
                           }
                         ]
    }
]
```

How do I do that? I may also want to add custom fields to my API response JSON. 

Thanks!

**Comment by a119347** [comment]
config config_rest.yaml allowedActions
```
allowedActions:
  - index
  - show(function name)
```
and then write the show function to show your particular record

**Comment by SaifurRahmanMohsin** [comment]
Hey. Sorry for delayed response. I'm a little held up so haven't checked this space. @a119347 is right. The show action must be used for selecting single record and not index. Unfortunately, I haven't included the default behavior for the show action in the current version of the plugin but you can write a custom function for now to solve this (see above comment). Read the laravel documentation on restful controllers to see how. Alternatively, wait a couple of days and I'll add the rest of the default behaviours as an update. I have a plan to add several features:

- [ ] Write default behaviours for usual actions: show, delete, update, etc.
- [ ] Update restConfig structure to allow field exceptions, specify transformers
- [ ] Build UI component to write REST routes directly from frontend than routes.php

So I do have plans to update the plugin to add several features but I'm working on several other plugins and projects so updates will take time. Hope you can understand. I'll notify you here once the update is done and then close the issue after.

**Comment by MilosStanic** [comment]
Hi, and thank you both for your answers. 

I have, unfortunately, purchased one other RESTful plugin, but it wasn't able to access models via controllers, only tables. 

I wish to pull models converted to JSON, including relationships. So this other plugin was not useful for me. 

Looking forward to seeing update of your plugin.
Thanks again.

**Comment by SaifurRahmanMohsin** [comment]
Hey @MilosStanic. Kindly check the current version of the plugin 1.0.3 as I have added support for relations. In the case of your example, this is how the Rest Behaviour config file show look like…

```
# ===================================
#  Rest Behavior Config
# ===================================

# Allowed Rest Actions
allowedActions:
  show:
      relations:
        - questions

# Model Class name
modelClass: Acme\YourPlugin\Models\Test
```

And kindly make another issue here should you face any issues so that I can fix it.
